src/storages/pluggable/index.ts

```diff
@@ -0,0 +1,67 @@
+import { ICustomStorageWrapper, IStorageAsync, IStorageFactoryParams } from '../types';
+
+import KeyBuilderSS from '../KeyBuilderSS';
+import { SplitsCachePluggable } from './SplitsCachePluggable';
+import { SegmentsCachePluggable } from './SegmentsCachePluggable';
+import { ImpressionsCachePluggable } from './ImpressionsCachePluggable';
+import { EventsCachePluggable } from './EventsCachePluggable';
+
+import { SDK_SPLITS_ARRIVED, SDK_SEGMENTS_ARRIVED } from '../../readiness/constants';
+import { wrapperAdapter, METHODS_TO_PROMISE_WRAP } from './wrapperAdapter';
+import { isObject } from '../../utils/lang';
+
+export interface PluggableStorageOptions {
+  prefix?: string
+  wrapper: ICustomStorageWrapper
+}
+
+/**
+ * Validate pluggable storage factory options.
+ *
+ * @param options user options
+ * @throws Will throw an error if the options are invalid. Example: wrapper is not provided or doesn't have some methods.
+ */
+function validatePluggableStorageOptions(options: any) {
+  if (!isObject(options) || !isObject(options.wrapper)) throw new Error('No `wrapper` option was provided');
+
+  const wrapper = options.wrapper;
+  const missingMethods = METHODS_TO_PROMISE_WRAP.filter(method => typeof wrapper[method] !== 'function');
+  if (missingMethods.length) throw new Error(`Wrapper instance must implement the following methods: ${missingMethods}`);
+}
+
+/**
+ * Pluggable storage factory for consumer server-side & client-side SplitFactory.
+ */
+export function PluggableStorage(options: PluggableStorageOptions) {
+
+  validatePluggableStorageOptions(options);
+
+  const prefix = options.prefix ? options.prefix + '.SPLITIO' : 'SPLITIO';
+
+  return function PluggableStorageFactory({ log, metadata, readinessManager }: IStorageFactoryParams): IStorageAsync {
+
```

Comment:
Remove this empty line

Reply:
done! something else to add to the linter